src/runtime_src/xdp/profile/plugin/aie_profile/aie_plugin.cpp

```diff
@@ -1196,87 +1195,157 @@ namespace xdp {
 
-      if (XAIE_CORE_MOD == mod || XAIE_MEM_MOD == mod) {
-        if (3 != metrics[i].size()) {
-          continue;
-        }
-        for (size_t j = 0; j < metrics[i].size(); ++j) {
-          boost::replace_all(metrics[i][j], "{", "");
-          boost::replace_all(metrics[i][j], "}", "");
+      std::vector<tile_type> tiles;
+
+      if (3 != metrics[i].size()) {
+        continue;
+      }
+
+      for (size_t j = 0; j < metrics[i].size(); ++j) {
+        boost::replace_all(metrics[i][j], "{", "");
+        boost::replace_all(metrics[i][j], "}", "");
+      }
+      std::vector<std::string> minTile;
+      boost::split(minTile, metrics[i][0], boost::is_any_of(","));
+      uint32_t minCol = std::stoi(minTile[0]);
+      uint32_t minRow = std::stoi(minTile[1]);
+
+      std::vector<std::string> maxTile;
+      boost::split(maxTile, metrics[i][1], boost::is_any_of(","));
+      uint32_t maxCol = std::stoi(maxTile[0]);
+      uint32_t maxRow = std::stoi(maxTile[1]);
+
+      for (uint32_t col = minCol; col <= maxCol; ++col) {
+        for (uint32_t row = minRow; row <= maxRow; ++row) {
+          xrt_core::edge::aie::tile_type tile;
+          tile.col = col;
+          tile.row = row;
+          tiles.push_back(tile);
         }
-        std::vector<std::string> minTile;
-        boost::split(minTile, metrics[i][0], boost::is_any_of(","));
-        uint32_t minCol = std::stoi(minTile[0]);
-        uint32_t minRow = std::stoi(minTile[1]);
-
-        std::vector<std::string> maxTile;
-        boost::split(maxTile, metrics[i][1], boost::is_any_of(","));
-        uint32_t maxCol = std::stoi(maxTile[0]);
-        uint32_t maxRow = std::stoi(maxTile[1]);
-  
-        for (uint32_t col = minCol; col <= maxCol; ++col) {
-          for (uint32_t row = minRow; row <= maxRow; ++row) {
-            xrt_core::edge::aie::tile_type tile;
-            tile.col = col;
-            tile.row = row;
-//            tiles.push_back(tile);
-            mConfigMetrics[moduleIdx][tile] = metrics[i][2];
-            // check if same tiles are recognized
+      }
+      if (XAIE_PL_MOD == mod) {
+        // Check and set mem_row/column
+
+        int plioCount = 0;
+        auto plios = xrt_core::edge::aie::get_plios(device.get());
+        for (auto& plio : plios) {
+          auto isMaster = plio.second.slaveOrMaster;
+          auto streamId = plio.second.streamId;
+
+          // If looking for specific ID, make sure it matches
+          if ((mChannelId >= 0) && (mChannelId != streamId))
+            continue;
+
+          // Make sure it's desired polarity
+          // NOTE: input = slave (data flowing from PLIO)
+          //       output = master (data flowing to PLIO)
+          if ((isMaster && (metrics[i][2] == "input_bandwidths"))
+              || (isMaster && (metrics[i][2] == "input_stalls_idle"))
+              || (!isMaster && (metrics[i][2] == "output_bandwidths"))
+              || (!isMaster && (metrics[i][2] == "output_stalls_idle")))
```

Comment:
These are not the correct metric set names as the "*_idle" ones do not exist anymore.

Reply:
I have removed the comparison with *_idle metrics.